frontend/public/module/k8s/k8s-actions.js

```diff
@@ -155,34 +153,43 @@ const actions = {
     dispatch({type: types.watchK8sList, id, query});
     REF_COUNTS[id] = 1;
 
-    // Fetch entire list (XHR) then use its resourceVersion to
-    //  start listening on a WS (?resourceVersion=$resourceVersion)
-    //  start the process over when:
-    //   1. the WS closes abnormally
-    //   2. the WS can not establish a connection within $TIMEOUT
+    /** @type {(continueToken: string) => Promise<string>} */
+    const incrementallyLoad = async(continueToken = '') => {
+      const response = await k8sList(k8skind, {...query, limit: paginationLimit, ...(continueToken ? {continue: continueToken} : {})}, true);
+
+      if (response.metadata.continue) {
+        dispatch(actions.bulkAddToList(id, response.items));
```

Comment:
It would be nice to check `REF_COUNTS[id]` in here and abort the loading if it's not > 0. That way we're not trying to load data long after a component has been unmounted.

Reply:
see the [hidden discussion above](https://github.com/openshift/console/pull/5#discussion_r189729865)

Reply:
I see. I'm also OK with fixing this in a follow-up PR.